sycl/source/detail/platform_util.cpp

```diff
@@ -62,21 +72,49 @@ uint32_t PlatformUtil::getMaxClockFrequency() {
   Buff = Buff.substr(Buff.rfind(' '), Buff.length());
   Freq *= std::stod(Buff);
   return Freq;
+#else
+#warning Your platform is not supported!
```

Comment:
This is going to produce warning any time when code is compiled for non-x86 platform, right? But patch enables the support of non-x64 platform, so this is a bit confusing.

Reply:
The warning is meant to indicate that the specific function is not supported.  Should I change the warning?  It is very hard to determine the CPU frequency on an ARM platform and the only thing I found requires a Linux 5.x kernel and read access to `/sys`...

Reply:
I see, I am ok to not support this function for ARM. 
If someone compiles this for ARM and calls this function 0 is going to be returned, should we also throw runtime error or compile-time warning is enough?

Reply:
Currently, this programs throws the following error unconditionally, but I figure that will be removed at some point so I decided to add a placeholder for a non-x86 implementation, for whenever somebody figures out how to get the frequency of an ARM CPU.
```c++
throw runtime_error(
      "max_clock_frequency parameter is not supported for host device",	
      PI_INVALID_DEVICE);
```